README.md

```diff
@@ -1,3 +1,150 @@
+
+## Set up the project
+
+If you don't want to copy the code from this document, you can clone the repository and use the files in there.
+   
+1. Create a new folder for your Python scripts.
+
+2. In that folder, create two files named `pub.py` and `sub.py`.
+
+3. Install the PubNub Python SDK using `pip`:
+
+    ```text
+    pip install 'pubnub>=4.6.0'
+    ```
+
+## Add project files
+
+If you don't want to copy the code from this document, you can clone the repository and use the files in there.
+
+1. Open the `pub.py` file and add the following code. Remember to also replace the _myPublishKey_ and _mySubscribeKey_ placeholders with your keys.
+
+    ```python
+    from pubnub.callbacks import SubscribeCallback
+    from pubnub.enums import PNStatusCategory
+    from pubnub.pnconfiguration import PNConfiguration
+    from pubnub.pubnub import PubNub
+
+    ENTRY = "Earth"
+    CHANNEL = "the_guide"
+    the_update = None
+
+    # replace the key placeholders with your own PubNub publish and subscribe keys
+    pnconfig = PNConfiguration()
+    pnconfig.publish_key = "myPublishKey"
+    pnconfig.subscribe_key = "mySubscribeKey"
+    pnconfig.uuid = "serverUUID-PUB"
+
+    pubnub = PubNub(pnconfig)
+
+    print
+    print("*****************************************")
+    print("* Submit updates to The Guide for Earth *")
+    print("*     Enter 42 to exit this process     *")
+    print("*****************************************")
+
+    while the_update != "42":
+        print
+        the_update = input("Enter an update for Earth: ")
+        the_message = {"entry": ENTRY, "update": the_update}
+        envelope = pubnub.publish().channel(CHANNEL).message(the_message).sync()
+
+        if envelope.status.is_error():
+            print("[PUBLISH: fail]")
+            print("error: %s" % envelope.status.error)
+            pass
+        else:
+            print("[PUBLISH: sent]")
+            print("timetoken: %s" % envelope.result.timetoken)
+            pass
+    ```
+
+2. Open the `sub.py` file and add the following code. Remember to also replace the _myPublishKey_ and _mySubscribeKey_ placeholders with your keys.
+
+    ```python
+    import os
+
+    from pubnub.callbacks import SubscribeCallback
+    from pubnub.enums import PNStatusCategory
+    from pubnub.pnconfiguration import PNConfiguration
+    from pubnub.pubnub import PubNub
+
+    ENTRY = "Earth"
+    CHANNEL = "the_guide"
+
+    # replace the key placeholders with your own PubNub publish and subscribe keys
+    pnconfig = PNConfiguration()
+    pnconfig.publish_key = "myPublishKey"
+    pnconfig.subscribe_key = "mySubscribeKey"
+    pnconfig.uuid = "serverUUID-SUB"
+
+    pubnub = PubNub(pnconfig)
+
+    class MySubscribeCallback(SubscribeCallback):
+        def message(self, pubnub, event):
+            print("[MESSAGE received]")
+
+            if event.message["update"] == "42":
+                print("The publisher has ended the session.")
+                os._exit(0)
```

Comment:
Do we have to use os_exit(0) here?

Reply:
I think so, the script should close both scripts `pub` and `sub`.